packages/protocol/test/governance/double_signing_slasher.ts

```diff
@@ -0,0 +1,160 @@
+    mockLockedGold = await MockLockedGold.new()
+    registry = await Registry.new()
+    validators = await MockValidators.new()
+    slasher = await DoubleSigningSlasher.new()
+    await accountsInstance.initialize(registry.address)
+    await registry.setAddressFor(CeloContractName.Accounts, accountsInstance.address)
+    await registry.setAddressFor(CeloContractName.LockedGold, mockLockedGold.address)
+    await registry.setAddressFor(CeloContractName.Validators, validators.address)
+    await validators.affiliate(group, { from: validator })
+    await validators.affiliate(accounts[3], { from: accounts[4] })
+    await slasher.initialize(registry.address, 10000, 100)
+    await Promise.all(
+      accounts.map((account) => mockLockedGold.incrementNonvotingAccountBalance(account, 50000))
+    )
+  })
+
+  describe('#initialize()', () => {
+    it('should have set the owner', async () => {
+      const owner: string = await slasher.owner()
+      assert.equal(owner, accounts[0])
+    })
+    it('should have set slashing incentives', async () => {
+      const res = await slasher.slashingIncentives()
+      assert.equal(res[0].toNumber(), 10000)
+      assert.equal(res[1].toNumber(), 100)
+    })
+    it('can only be called once', async () => {
+      await assertRevert(slasher.initialize(registry.address, 10000, 100))
+    })
+  })
+
+  describe('#setSlashingIncentives()', () => {
+    it('can only be set by the owner', async () => {
+      await assertRevert(slasher.setSlashingIncentives(123, 67, { from: nonOwner }))
+    })
+    it('reward cannot be larger than penalty', async () => {
+      await assertRevert(slasher.setSlashingIncentives(123, 678))
+    })
+    it('should have set slashing incentives', async () => {
+      await slasher.setSlashingIncentives(123, 67)
+      const res = await slasher.slashingIncentives()
+      assert.equal(res[0].toNumber(), 123)
+      assert.equal(res[1].toNumber(), 67)
+    })
+    it('should emit the corresponding event', async () => {
+      const resp = await slasher.setSlashingIncentives(123, 67)
+      assert.equal(resp.logs.length, 1)
+      const log = resp.logs[0]
+      assertContainSubset(log, {
+        event: 'SlashingIncentivesSet',
+        args: {
+          penalty: new BigNumber(123),
+          reward: new BigNumber(67),
+        },
+      })
+    })
+  })
+
+  describe('#slash()', () => {
+    const blockNumber = 100
+    const validatorIndex = 5
+    const blockA = ['0x12', '0x12', '0x12']
+    const blockB = ['0x13', '0x13', '0x13']
+    const blockC = ['0x11', '0x13', '0x14']
+    beforeEach(async () => {
+      await slasher.setBlockNumber(blockA, blockNumber)
+      await slasher.setBlockNumber(blockB, blockNumber + 1)
+      await slasher.setBlockNumber(blockC, blockNumber)
+      await slasher.setEpochSigner(Math.floor(blockNumber / EPOCH), validatorIndex, validator)
+      await slasher.setEpochSigner(Math.floor(blockNumber / EPOCH), validatorIndex + 1, validator)
```

Comment:
Why are we setting `validator` to be the signer at two different indices in the same epoch? Should this be for different epochs?